usr/share/rear/rescue/default/500_ssh.sh

```diff
@@ -1,65 +1,61 @@
+# files without a [] are mandatory.
 
-    COPY_AS_IS=( "${COPY_AS_IS[@]}" /etc/ssh* /root/.s[s]h /root/.shos[t]s )
-    PROGS=(
-    ${PROGS[@]}
-    ssh sshd scp sftp
+COPY_AS_IS=( "${COPY_AS_IS[@]}" /etc/ssh* /etc/openssh* /etc/centrifydc/ssh* /root/.s[s]h /root/.shos[t]s )
+PROGS=(
+    "${PROGS[@]}"
+    ssh sshd scp sftp ssh-agent ssh-keygen
     $(
-        read subsys sftp file junk < <( grep sftp /etc/sshd_co[n]fig /etc/ssh/sshd_co[n]fig /etc/openssh/sshd_co[n]fig /etc/centrifydc/ssh/sshd_co[n]fig 2>/dev/null )
-        echo $file
-    )
+        read subsys sftp original_file junk < <( grep sftp /etc/sshd_co[n]fig /etc/ssh/sshd_co[n]fig /etc/openssh/sshd_co[n]fig /etc/centrifydc/ssh/sshd_co[n]fig 2>/dev/null )
+        echo $original_file
     )
+)
 
-    # we need to add some specific NSS lib for shadow passwords to work on RHEL 6/7
-    LIBS=( ${LIBS[@]} /usr/lib64/libfreeblpriv3.* /lib/libfreeblpriv3.* )
-    Log "Adding required libfreeblpriv3.so to LIBS"
+# Part 1: SSH server (sshd) - this is for logging into the recovery system via SSH
 
-    # copy ssh user
-    # getent will return all entries that match the key(s) exactly - most systems use 'sshd', some may use 'ssh', none should use both.
-    # Only the first line (first returned entry) will be used by 'read' in 'IFS=: read ... <<<"$PASSWD_SSH"', so we ask for sshd first.
-    PASSWD_SSH=$(getent passwd sshd ssh)
-    if test -n "$PASSWD_SSH" ; then
-    # sshd:x:71:65:SSH daemon:/var/lib/sshd:/bin/false
-        IFS=: read user ex uid gid gecos homedir junk <<<"$PASSWD_SSH"
-	# skip if this user exists already in the restore system
-	if ! egrep -q "^$user:" $TARGET_FS_ROOT/etc/passwd ; then
-		echo "$PASSWD_SSH" >>$ROOTFS_DIR/etc/passwd
-	fi
-        # add ssh group to be collected later
-        CLONE_GROUPS=( "${CLONE_GROUPS[@]}" "$gid" )
-        mkdir -p $v -m 0700 "$ROOTFS_DIR$homedir" >&2
-        chown $v root.root "$ROOTFS_DIR$homedir" >&2
-    fi
+# we need to add some specific NSS lib for shadow passwords to work on RHEL 6/7
+LIBS=( "${LIBS[@]}" /usr/lib64/libfreeblpriv3.* /lib/libfreeblpriv3.* )
+Log "Adding required libfreeblpriv3.so to LIBS"
 
-    echo "ssh:23:respawn:/bin/sshd -D" >>$ROOTFS_DIR/etc/inittab
+# copy ssh user
+# getent will return all entries that match the key(s) exactly - most systems use 'sshd', some may use 'ssh', none should use both.
+# CLONE_USERS will also automatically clone the users' primary group
+# Only the first line (first returned entry) will be used by 'read' in 'IFS=: read ... <<<"$PASSWD_SSH"', so we ask for sshd first.
+PASSWD_SSH=$(getent passwd sshd ssh)
+if test "$PASSWD_SSH" ; then
+    # sshd:x:71:65:SSH daemon:/var/lib/sshd:/bin/false
+    IFS=: read user ex uid gid gecos homedir junk <<<"$PASSWD_SSH"
+    CLONE_USERS=( "${CLONE_USERS[@]}" $user )
+    DIRECTORIES_TO_CREATE=( "${DIRECTORIES_TO_CREATE[@]}" "$homedir 0700 root root")
```

Comment:
I am so **stupid**. `DIRECTORIES_TO_CREATE` targets the recovered system and **not** the rescue system. I'll rework this PR, @rear/contributors FYI that you won't step into the same mental trap.

Reply:
@schlomo 
don't worry - I also get often confused what belongs
to creating the recovery system and what belongs to
re-creating the target system. That thingy should be
better named DIRECTORIES_TO_RECREATE but
we cannot do that with reasonable effort because
it is a user config variable.

Reply:
@schlomo 
wait!
I introduced DIRECTORIES_TO_CREATE on Aug 25 2017 via
https://github.com/rear/rear/commit/d27e424399006607e1163ced1d5fc06afa9a50ea
which was after the ReaR 2.2 release so that I can
"just rename" it into DIRECTORIES_TO_RECREATE
right now... :-)

Reply:
@schlomo 
I have a problem with the naming
DIRECTORIES_TO_CREATE
versus
DIRECTORIES_TO_RECREATE
because the latter is technically wrong.
All what there is in DIRECTORIES_TO_CREATE
gets created in the target system regardless what
there was before in the original system,
see its description in default.conf.
This is the reason why I had intentionally
named it ..._CREATE and not ..._RECREATE.
But on the other hand ..._RECREATE could
make it more obvious that this thingy belongs
to the target system and not to the recovery system.
What name do you prefer?

Reply:
How about `DIRECTORY_ENTRIES_TO_RESTORE` to denote that ➊ this is about restoring data into the target system and ➋ that this only restores the directory entries but not the content.

Reply:
Any meaningful and non-confusing name you wish!
I.e. DIRECTORY_ENTRIES_TO_RESTORE is o.k. for me
regardless that technically those are not 're-stored' from
any previously made 'store' but just created from scratch.

Reply:
Yes, I was hoping the "restore" makes it clear that this is the opposite of "backup". What do you think about `DIRECTORY_ENTRIES_TO_RECOVER` in an allusion to `rear recover`?

Reply:
Good idea!
I will rename it into DIRECTORY_ENTRIES_TO_RECOVER.